packages/flutter_tools/lib/src/ios/simulators.dart

```diff
@@ -361,6 +365,17 @@ class IOSSimulator extends Device {
   }
 
   @override
+  bool isAppInstalled(ApplicationPackage app) {
+    return exitsHappy(<String>[
+      'xcrun',
+      'simctl',
+      'get_app_container',
```

Comment:
You could add this as a new method to `SimControl`; that class already wraps a lot of `simctl` commands.